src/robot/running/arguments/argumentconverter.py

```diff
@@ -59,3 +65,8 @@ def _get_type(self, name, value):
         if name in self._argspec.defaults:
             return type(self._argspec.defaults[name]), False
         return None, None
+
+    def _need_conversion(self, value):
+        if self._dry_run and VARIABLE_PATTERN.search(value):
```

Comment:
We actually have `robot.variables.contains_var` utility that could be used here instead. It might even make this `_need_conversion` helper unnecessary.

Reply:
Thank you for pointing this out. I didn't know about `contains_var` utility. This is much better than using a custom regex.